File: Algorithms/989.数组形式的整数加法/addToArrayForm.java
class Solution {
    public List<Integer> addToArrayForm(int[] A, int K) {
        List<Integer> res = new ArrayList<Integer>();
        int n = A.length;
        for (int i = n - 1; i >= 0 || K > 0; --i, K /= 10) {
            if (i >= 0)
                K += A[i];
            res.add(K % 10);
        }
        Collections.reverse(res);
        return res;
    }
}

File: Algorithms/989/addToArrayForm.py
class Solution:
    def addToArrayForm(self, A: List[int], K: int) -> List[int]:
        A.reverse()
        K = list(map(int, reversed(str(K))))
        if len(A) < len(K):
            A, K = K, A
        carry = 0
        for i in range(len(K)):
            carry, A[i] = divmod(A[i] + K[i] + carry, 10)
        i = len(K)
        while carry and i < len(A):
            carry, A[i] = divmod(A[i] + carry, 10)
            i += 1
        if carry:
            A.append(carry)
        A.reverse()
        return A
